src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -140,12 +141,39 @@ public ResultWrapper<Address> eth_coinbase()
             return ResultWrapper<UInt256?>.Success(0);
         }
 
-        [Todo("Gas pricer to be implemented")]
         public ResultWrapper<UInt256?> eth_gasPrice()
         {
-            return ResultWrapper<UInt256?>.Success(20.GWei());
+            Block? headBlock = _blockFinder.FindHeadBlock();
+            ThrowExceptionIfHeadBlockIsNull(headBlock);
+            BlockNumberToBlockDictionary = CreateBlockNumberToBlockDictionary(headBlock);
+            return GasPriceOracle!.GasPriceEstimate(headBlock, BlockNumberToBlockDictionary);
+        }
+
+        private void ThrowExceptionIfHeadBlockIsNull(Block? headBlock)
+        {
+            if (headBlock == null)
+            {
+                throw new Exception("Head Block was not found.");
+            }
         }
 
+        private Dictionary<long, Block> CreateBlockNumberToBlockDictionary(Block? headBlock)
```

Comment:
I think we should move everything to GasPriceOracle, but you can ignore it..
But I wonder if we need to build this dict everytime? What geth is doing here?